jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -91,15 +91,40 @@ class SpectrumPerSpaxel(ProfileFromCube):
     action_text = 'See spectrum at a single spaxel'
     tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_moving = False
+
+    def activate(self):
+        super().activate()
+        for k in ("y_min", "y_max"):
+            self._profile_viewer.state.add_callback(k, self.on_limits_change)
+
+    def deactivate(self):
+        for k in ("y_min", "y_max"):
+            self._profile_viewer.state.remove_callback(k, self.on_limits_change)
+        super().deactivate()
+
+    def on_limits_change(self, *args):
```

Comment:
I purposely implement this only for Cubeviz and not its parent class because I have no idea what downstream where is using the parent class and want to keep my scope to fix Cubeviz only.